lib/ansible/modules/unarchive.py

```diff
@@ -383,8 +383,12 @@ def files_in_archive(self):
         return self._files_in_archive
 
     def is_unarchived(self):
-        # BSD unzip doesn't support zipinfo listings with timestamp.
-        cmd = [self.zipinfocmd_path, '-T', '-s', self.src]
+        if self.zipinfocmd_path:
+            # BSD unzip doesn't support zipinfo listings with timestamp.
+            cmd = [self.zipinfocmd_path, '-T', '-s', self.src]
+        else:
+            cmd = [self.cmd_path, '-Z', self.src]
```

Comment:
@bcoca Could you please test if `unzip -Z <archive>` works on BSD? I don't have BSD machines handy.

Reply:
Doesn't look like it.
```
root@freebsd:~$ unzip -Z test.zip
Zipinfo mode needs additional options
```

Reply:
Only `Z1` mode is supported, so this works.

```
root@freebsd:~$ unzip -Z1 test.zip
1.txt
2.txt
```

Reply:
It won't be enough, we need something like - 
```
unzip -Z jce_policy-6.zip
Archive:  jce_policy-6.zip
Zip file size: 9101 bytes, number of entries: 5
drwxr-xr-x  2.2 unx        0 bx stor 06-Nov-17 07:40 jce/
-r--r--r--  2.2 unx     2663 tx defN 06-Nov-17 07:40 jce/COPYRIGHT.html
-r--r--r--  2.2 unx     8386 tx defN 06-Nov-17 07:40 jce/README.txt
-rw-r--r--  2.2 unx     2465 bx defN 06-Nov-17 07:40 jce/US_export_policy.jar
-rw-r--r--  2.2 unx     2481 bx defN 06-Nov-17 07:40 jce/local_policy.jar
5 files, 15995 bytes uncompressed, 8369 bytes compressed:  47.7%
```

Reply:
Yeah, then we'll have to fail if `zipinfo` is missing then.